source/fuzz/fuzzer_util.cpp

```diff
@@ -1616,6 +1616,38 @@ bool IdUseCanBeReplaced(opt::IRContext* ir_context,
     return false;
   }
 
+  switch (use_instruction->opcode()) {
+    case SpvOpAtomicLoad:
+    case SpvOpAtomicStore:
+    case SpvOpAtomicExchange:
+    case SpvOpAtomicCompareExchange:
+    case SpvOpAtomicCompareExchangeWeak:
+    case SpvOpAtomicIIncrement:
+    case SpvOpAtomicIDecrement:
+    case SpvOpAtomicIAdd:
+    case SpvOpAtomicISub:
+    case SpvOpAtomicSMin:
+    case SpvOpAtomicUMin:
+    case SpvOpAtomicSMax:
+    case SpvOpAtomicUMax:
+    case SpvOpAtomicAnd:
+    case SpvOpAtomicOr:
+    case SpvOpAtomicXor:
+
+      // use_in_operand_index Should be the replaced by synonymous_id (I think
+      // so)??
+      if (ir_context->get_def_use_mgr()
+                  ->GetDef(use_in_operand_index)
+                  ->opcode() != SpvOpConstant &&
+          (use_in_operand_index == 1 || use_in_operand_index == 2)) {
+        return false;
+      }
+      break;
```

Comment:
What we want to do is return `false` for any operand that corresponds to a memory scope or memory semantics argument.

This varies from instruction to instruction. For many/most of them, it is indeed input operands 1 and 2, but not for all of them. For example, for `OpAtomicCompareExchange`, input operands 1, 2 and 3 cannot be replaced. Please check each instruction carefully.

Also: we should simply return false in such cases - we don't need to check whether the existing operand is an SpvOpConstant.

Reply:
Done.